pkg/resources/connection.go

```diff
@@ -0,0 +1,313 @@
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		currentAccountIdentifier := sdk.NewAccountIdentifier(sessionDetails.OrganizationName, sessionDetails.AccountName)
+
+		enableFailoverToAccounts := make([]string, 0)
+		for _, allowedAccount := range connection.FailoverAllowedToAccounts {
+			allowedAccountIdentifier := sdk.NewAccountIdentifierFromFullyQualifiedName(allowedAccount)
+			if currentAccountIdentifier.FullyQualifiedName() == allowedAccountIdentifier.FullyQualifiedName() {
+				continue
+			}
+			enableFailoverToAccounts = append(enableFailoverToAccounts, allowedAccountIdentifier.Name())
+		}
+
+		if len(enableFailoverToAccounts) == 0 {
+			err := d.Set("enable_failover_to_accounts", []any{})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		} else {
+			err := d.Set("enable_failover_to_accounts", enableFailoverToAccounts)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		return nil
+	}
+}
+
+func UpdateContextConnection(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseAccountObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	connectionSetRequest := new(sdk.ConnectionSetRequest)
+	connectionUnsetRequest := new(sdk.ConnectionUnsetRequest)
+
+	if d.HasChange("enable_failover_to_accounts") {
+		before, after := d.GetChange("enable_failover_to_accounts")
+
+		getFailoverToAccounts := func(failoverConfig []any) []sdk.AccountIdentifier {
+			failoverEnabledToAccounts := make([]sdk.AccountIdentifier, 0)
+
+			for _, allowedAccount := range failoverConfig {
+				accountIdentifier := sdk.NewAccountIdentifierFromFullyQualifiedName(allowedAccount.(string))
+				failoverEnabledToAccounts = append(failoverEnabledToAccounts, accountIdentifier)
+			}
+			return failoverEnabledToAccounts
+		}
+
+		beforeFailover := getFailoverToAccounts(before.([]any))
+		afterFailover := getFailoverToAccounts(after.([]any))
+
+		addedFailovers, removedFailovers := ListDiff(beforeFailover, afterFailover)
+
+		if len(addedFailovers) > 0 {
+			err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).
+				WithEnableConnectionFailover(*sdk.NewEnableConnectionFailoverRequest().
+					WithToAccounts(addedFailovers),
+				),
+			)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		if len(removedFailovers) > 0 {
+			err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).
+				WithDisableConnectionFailover(*sdk.NewDisableConnectionFailoverRequest().
+					WithToAccounts(*sdk.NewToAccountsRequest().
+						WithAccounts(removedFailovers),
+					),
+				),
+			)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	if d.HasChange("is_primary") {
+		if is_primary := d.Get("is_primary").(bool); is_primary {
+			err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).WithPrimary(is_primary))
```

Comment:
When `is_primary` is removed or set to false, this branch won't execute. We should handle this other case (maybe with forcenew, since we don't have an option to revert `PRIMARY` in `ALTER`).

You can use our stringbools, take a look at handling `is_secret` in https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/resources/view.go

Reply:
after splitting i've added a custom diff on secondary_connection resource, which detects if the new value has been changed externally to `true` if so, we should re-create the secondary_connection to still serve as secondary, not primary one.